src/driver/amdxdna/aie2_ctx.c

```diff
@@ -1065,12 +1065,18 @@ int aie2_cmd_wait(struct amdxdna_ctx *ctx, u64 seq, u32 timeout)
 
 void aie2_hmm_invalidate(struct amdxdna_gem_obj *abo, unsigned long cur_seq)
 {
-	struct amdxdna_dev *xdna = to_xdna_dev(to_gobj(abo)->dev);
 	struct drm_gem_object *gobj = to_gobj(abo);
 	long ret;
 
-	ret = dma_resv_wait_timeout(gobj->resv, DMA_RESV_USAGE_BOOKKEEP,
-				    true, MAX_SCHEDULE_TIMEOUT);
-	if (!ret || ret == -ERESTARTSYS)
-		XDNA_ERR(xdna, "Failed to wait for bo, ret %ld", ret);
+	/*
+	 * Must wait forever, otherwise, memory was unmapped then FW might crash.
+	 * In case FW not response, TDR will terminal context execution and unref all BOs.
+	 * This loop will eventually exit.
+	 */
+	while (1) {
+		ret = dma_resv_wait_timeout(gobj->resv, DMA_RESV_USAGE_BOOKKEEP,
+					    true, MAX_SCHEDULE_TIMEOUT);
```

Comment:
Instead of while (1), I suggest dma_resv_wait_timeout(..., false, MAX_SCHEDULE_TIMEOUT);